docs/developer_workflow.md

```diff
@@ -1,43 +1,141 @@
 # Developer workflow
 
-## Download source code
-<pre>
-git clone https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix.git
-cd cluster-api-provider-nutanix
-</pre>
-
-## Build source code
-<pre>
-make docker-build
-</pre>
-
-## Create test management cluster
-<pre>
-make kind-create
-</pre>
+This document outlines how to
+
+## Build the source code
+
+1. Download source code:
+
+    ```shell
+    git clone https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix.git
+    cd cluster-api-provider-nutanix
+    ```
+
+1. Build a container image with the source code:
+
+    ```shell
+    make docker-build
+    ```
+
+## Create a local management cluster
+
+1. Create a [KIND](https://kind.sigs.k8s.io/) cluster:
+
+   ```shell
+    make kind-create
+    ```
+
+This will configure [kubectl](https://kubernetes.io/docs/reference/kubectl/) for the local cluster.
+
+## Build a CAPI image
+
+1. Follow the instructions [here](https://image-builder.sigs.k8s.io/capi/providers/nutanix.html#building-capi-images-for-nutanix-cloud-platform-ncp) to build a CAPI image.
+   The image name printed at the end of the build will be needed to create a Kubernetes cluster.
 
 ## Prepare local clusterctl
-<pre>
-make prepare-local-clusterctl
-</pre>
-
-## Deploy cluster-api-provider-nutanix CRDs on test management cluster
-<pre>
-make deploy
-</pre>
-
-## Deploy test workload cluster
-Note: Update ./clusterctl.yaml with appropriate configuration before running following commands
-<pre>
-make test-clusterctl-create
-</pre>
-
-## Get test workload cluster kubeconfig
-<pre>
-make test-kubectl-workload
-</pre>
-
-## Delete test workload cluster
-<pre>
-make test-clusterctl-delete
-</pre>
\ No newline at end of file
+
+1. Make a copy of the [clusterctl](https://cluster-api.sigs.k8s.io/clusterctl/overview) configuration file:
+
+    ```shell
+    cp -n clusterctl.yaml.tmpl clusterctl.yaml
```

Comment:
`make prepare-local-clusterctl` copies ./clusterct.yaml from the local directory. Asking users to modify this file in the repo which could potentially have creds in it seemed dangerous.

Alternatively we can document to update `cat ~/.cluster-api/clusterctl.yaml`.

Reply:
with clusterctl.yaml added in `.gitignore` in this PR, is there any reason for the concern? We can, ofc, move it to a home-scoped hidden directory as you propose.

Reply:
With this PR I renamed what used to be `clusterctl.yaml` to `clusterctl.yaml.tmpl` and then asking devs to `cp` it back to `clusterctl.yaml` and edit the file. Then with the `.gitignore` there is no more risk of committing the creds.

This comment was really an explanation and editing `~/.cluster-api/clusterctl.yaml` was an alterative to the changes I made.
I prefer the changes I made of renaming  `clusterctl.yaml.tmpl` though.